### ISSUE TITLE
SPSearchContentSource fails Get when ContinuousCrawl equal to $true

### ISSUE THREAD

**Issue by luigilink** [PowerShell/SharePointDsc#827]
**Details of the scenario you try and problem that is occurring:**
When ContinuousCrawl parameter is defined to $true, the Get-Targetresource doesn't work:
```
Get-DscConfiguration : Échec de GetConfiguration.
Au caractère Ligne:1 : 1
+ Get-DscConfiguration -Verbose
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (MSFT_DSCLocalConfigurationManager:root/Microsoft/...gurationManager) [Get
   -DscConfiguration], CimException
    + FullyQualifiedErrorId : MI RESULT 1,Get-DscConfiguration
```

We have this event in event viewer, I think that the return of Get-SPDSCSearchCrawlSchedule function is not compliant with the input/output parameter:

```
Tâche {67CB3990-7559-11E8-810C-000D3A259A12} : 
Message Impossible d'effectuer un cast d'un objet de type 'System.String' en type 'System.Collections.IList'.
Nom du paramètre : value 
HResult : -2147024809 
StackTrack :    à Microsoft.Management.Infrastructure.Internal.Data.CimPropertyOfInstance.set_Value(Object value)
   à Microsoft.PowerShell.DesiredStateConfiguration.Internal.ResourceProviderAdapter.GetTargetResource(IntPtr resourceConfigurationInstanceHandle, IntPtr nonResourcePropetiesHandle, IntPtr metaConfigHandle, IntPtr regInstanceHandle, IStreamsHandler plugInStreamsHandler, IntPtr& outputInstanceHandle, IntPtr& errorInstanceHandle)
   à Microsoft.PowerShell.DesiredStateConfiguration.Internal.DscPluginManagedEntryWrapper.Get(IntPtr provContext, IntPtr instance, IntPtr nonResourcePropeties, IntPtr metaConfig, IntPtr regInstance, IntPtr outputInstance, IntPtr extendedError)
```

**The DSC configuration that is using the resource:**	
```
SPSearchContentSource APPLICATION_SpsSvcSearchContentSourceCRSE
{
  ContentSourceType = "SharePoint"
  Ensure = 'Present'
  ServiceAppName = 'SPSSCH'
  Addresses = 'http://sps15dev02'
  PsDscRunAsCredential = $credential
  ContinuousCrawl = $true
  Name =  'Content02'
  Priority = "Normal"
  CrawlSetting = "CrawlEverything"
}
```

**Version of the Operating System and PowerShell the DSC Target Node is running:**
Windows Server 2012 R2
WMF 5.1

**Version of SharePoint that is used (e.g. SharePoint 2016):**
SharePoint 2013

**Version of the DSC module you're using:**
2.3.0.0 (same issue with 2.0.0.0 to 2.2.0.0)

**Comment by johlju** [comment]
@luigilink Maybe there are a cast to an array missing somewhere when the value returned from something is a single string. Not looked at the code, just guessing. 🙂 Labeling it as a bug and help wanted.

**Comment by ykuijs** [comment]
@luigilink Can you run the following command on the SP server and let us know the outcome:
```
Invoke-DscResource -Name SPSearchContentSource -Method Get -Property @{
  ContentSourceType = "SharePoint"
  Ensure = 'Present'
  ServiceAppName = 'SPSSCH'
  Addresses = 'http://sps15dev02'
  PsDscRunAsCredential = $credential
  ContinuousCrawl = $true
  Name =  'Content02'
  Priority = "Normal"
  CrawlSetting = "CrawlEverything"
} -ModuleName SharePointDsc
```

**Comment by ykuijs** [comment]
@luigilink Any update?

**Comment by ykuijs** [comment]
I have been able to reproduce the error. Investigating now

**Comment by ykuijs** [comment]
Seems like this has something to do with the Addresses parameter. Your config only has one 

@luigilink Can you check what happens if you add another url to the content source (from Central Admin) and then run Get-DscConfiguration -Verbose once more?

**Comment by ykuijs** [comment]
Fix will be included in my next Bugfix PR

**Comment by ykuijs** [comment]
Closed via PR #838
